<!DOCTYPE html>
<!-- saved from url=(0046)https://jwestrob.github.io/Week_2_walkthrough/ -->
<html class="js no-touchevents cssanimations csstransitions" style=""><!--<![endif]--><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8">   <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1"> <title>Week 2 Walkthrough- Read Quality and Trimming – Jacob West-Roberts</title> <meta name="description" content="Moon is a minimal, one column jekyll theme for your blog."> <meta name="keywords" content="Read Quality"> <!-- Twitter Cards --> <meta name="twitter:card" content="summary"> <meta name="twitter:image" content="/assets/img/nerd.jpg"> <meta name="twitter:title" content="Week 2 Walkthrough- Read Quality and Trimming"> <meta name="twitter:description" content="Read Trimming and Quality Control!"> <meta name="twitter:site" content="@jwestrob"> <meta name="twitter:creator" content="@jwestrob"> <!-- Open Graph --> <meta property="og:locale" content="en_US"> <meta property="og:type" content="article"> <meta property="og:title" content="Week 2 Walkthrough- Read Quality and Trimming"> <meta property="og:description" content="Read Trimming and Quality Control!"> <meta property="og:url" content="/Week_2_walkthrough/"> <meta property="og:site_name" content="Jacob West-Roberts"> <meta property="og:image" content="/assets/img/nerd.jpg"> <link rel="canonical" href="https://jwestrob.github.io/Week_2_walkthrough/"> <link href="https://jwestrob.github.io/feed.xml" type="application/atom+xml" rel="alternate" title="Jacob West-Roberts Feed"> <!-- Handheld --> <meta name="HandheldFriendly" content="True"> <meta name="MobileOptimized" content="320"> <meta name="viewport" content="width=device-width, initial-scale=1.0"> <!-- CSS --> <link rel="stylesheet" href="./Week 2 Walkthrough- Read Quality and Trimming – Jacob West-Roberts_files/main.css"> <!-- JS --> <script src="./Week 2 Walkthrough- Read Quality and Trimming – Jacob West-Roberts_files/modernizr-3.3.1.custom.min.js"></script> <!-- Favicons --> <link rel="apple-touch-icon" href="https://jwestrob.github.io/assets/img/favicons/apple-icon-precomposed.png"> <link rel="apple-touch-icon" sizes="72x72" href="https://jwestrob.github.io/assets/img/favicons/apple-icon-72x72.png"> <link rel="apple-touch-icon" sizes="114x114" href="https://jwestrob.github.io/assets/img/favicons/apple-icon-114x114.png"> <link rel="apple-touch-icon" sizes="144x144" href="https://jwestrob.github.io/assets/img/favicons/apple-icon-144x144.png"> <link rel="shortcut icon" type="image/png" href="https://jwestrob.github.io/favicon.png"> <link rel="shortcut icon" href="https://jwestrob.github.io/favicon.ico"> <!-- Background Image --> <style type="text/css">body {background-image:url(/assets/img/pretty_tree_cropped_smaller.jpg); background-repeat: no-repeat; background-size: cover; }</style> <!-- Post Feature Image --> <style id="fit-vids-style">.fluid-width-video-wrapper{width:100%;position:relative;padding:0;}.fluid-width-video-wrapper iframe,.fluid-width-video-wrapper object,.fluid-width-video-wrapper embed {position:absolute;top:0;left:0;width:100%;height:100%;}</style><style>:is([id*='google_ads_iframe'],[id*='taboola-'],.taboolaHeight,.taboola-placeholder,#credential_picker_container,#credentials-picker-container,#credential_picker_iframe,[id*='google-one-tap-iframe'],#google-one-tap-popup-container,.google-one-tap-modal-div,#amp_floatingAdDiv,#ez-content-blocker-container) {display:none!important;min-height:0!important;height:0!important;}</style><style type="text/css">.MathJax_Hover_Frame {border-radius: .25em; -webkit-border-radius: .25em; -moz-border-radius: .25em; -khtml-border-radius: .25em; box-shadow: 0px 0px 15px #83A; -webkit-box-shadow: 0px 0px 15px #83A; -moz-box-shadow: 0px 0px 15px #83A; -khtml-box-shadow: 0px 0px 15px #83A; border: 1px solid #A6D ! important; display: inline-block; position: absolute}
.MathJax_Menu_Button .MathJax_Hover_Arrow {position: absolute; cursor: pointer; display: inline-block; border: 2px solid #AAA; border-radius: 4px; -webkit-border-radius: 4px; -moz-border-radius: 4px; -khtml-border-radius: 4px; font-family: 'Courier New',Courier; font-size: 9px; color: #F0F0F0}
.MathJax_Menu_Button .MathJax_Hover_Arrow span {display: block; background-color: #AAA; border: 1px solid; border-radius: 3px; line-height: 0; padding: 4px}
.MathJax_Hover_Arrow:hover {color: white!important; border: 2px solid #CCC!important}
.MathJax_Hover_Arrow:hover span {background-color: #CCC!important}
</style><style type="text/css">#MathJax_About {position: fixed; left: 50%; width: auto; text-align: center; border: 3px outset; padding: 1em 2em; background-color: #DDDDDD; color: black; cursor: default; font-family: message-box; font-size: 120%; font-style: normal; text-indent: 0; text-transform: none; line-height: normal; letter-spacing: normal; word-spacing: normal; word-wrap: normal; white-space: nowrap; float: none; z-index: 201; border-radius: 15px; -webkit-border-radius: 15px; -moz-border-radius: 15px; -khtml-border-radius: 15px; box-shadow: 0px 10px 20px #808080; -webkit-box-shadow: 0px 10px 20px #808080; -moz-box-shadow: 0px 10px 20px #808080; -khtml-box-shadow: 0px 10px 20px #808080; filter: progid:DXImageTransform.Microsoft.dropshadow(OffX=2, OffY=2, Color='gray', Positive='true')}
#MathJax_About.MathJax_MousePost {outline: none}
.MathJax_Menu {position: absolute; background-color: white; color: black; width: auto; padding: 5px 0px; border: 1px solid #CCCCCC; margin: 0; cursor: default; font: menu; text-align: left; text-indent: 0; text-transform: none; line-height: normal; letter-spacing: normal; word-spacing: normal; word-wrap: normal; white-space: nowrap; float: none; z-index: 201; border-radius: 5px; -webkit-border-radius: 5px; -moz-border-radius: 5px; -khtml-border-radius: 5px; box-shadow: 0px 10px 20px #808080; -webkit-box-shadow: 0px 10px 20px #808080; -moz-box-shadow: 0px 10px 20px #808080; -khtml-box-shadow: 0px 10px 20px #808080; filter: progid:DXImageTransform.Microsoft.dropshadow(OffX=2, OffY=2, Color='gray', Positive='true')}
.MathJax_MenuItem {padding: 1px 2em; background: transparent}
.MathJax_MenuArrow {position: absolute; right: .5em; padding-top: .25em; color: #666666; font-size: .75em}
.MathJax_MenuActive .MathJax_MenuArrow {color: white}
.MathJax_MenuArrow.RTL {left: .5em; right: auto}
.MathJax_MenuCheck {position: absolute; left: .7em}
.MathJax_MenuCheck.RTL {right: .7em; left: auto}
.MathJax_MenuRadioCheck {position: absolute; left: .7em}
.MathJax_MenuRadioCheck.RTL {right: .7em; left: auto}
.MathJax_MenuLabel {padding: 1px 2em 3px 1.33em; font-style: italic}
.MathJax_MenuRule {border-top: 1px solid #DDDDDD; margin: 4px 3px}
.MathJax_MenuDisabled {color: GrayText}
.MathJax_MenuActive {background-color: #606872; color: white}
.MathJax_MenuDisabled:focus, .MathJax_MenuLabel:focus {background-color: #E8E8E8}
.MathJax_ContextMenu:focus {outline: none}
.MathJax_ContextMenu .MathJax_MenuItem:focus {outline: none}
#MathJax_AboutClose {top: .2em; right: .2em}
.MathJax_Menu .MathJax_MenuClose {top: -10px; left: -10px}
.MathJax_MenuClose {position: absolute; cursor: pointer; display: inline-block; border: 2px solid #AAA; border-radius: 18px; -webkit-border-radius: 18px; -moz-border-radius: 18px; -khtml-border-radius: 18px; font-family: 'Courier New',Courier; font-size: 24px; color: #F0F0F0}
.MathJax_MenuClose span {display: block; background-color: #AAA; border: 1.5px solid; border-radius: 18px; -webkit-border-radius: 18px; -moz-border-radius: 18px; -khtml-border-radius: 18px; line-height: 0; padding: 8px 0 6px}
.MathJax_MenuClose:hover {color: white!important; border: 2px solid #CCC!important}
.MathJax_MenuClose:hover span {background-color: #CCC!important}
.MathJax_MenuClose:hover:focus {outline: none}
</style><style type="text/css">.MathJax_Preview .MJXf-math {color: inherit!important}
</style><style type="text/css">.MJX_Assistive_MathML {position: absolute!important; top: 0; left: 0; clip: rect(1px, 1px, 1px, 1px); padding: 1px 0 0 0!important; border: 0!important; height: 1px!important; width: 1px!important; overflow: hidden!important; display: block!important; -webkit-touch-callout: none; -webkit-user-select: none; -khtml-user-select: none; -moz-user-select: none; -ms-user-select: none; user-select: none}
.MJX_Assistive_MathML.MJX_Assistive_MathML_Block {width: 100%!important}
</style><style type="text/css">#MathJax_Zoom {position: absolute; background-color: #F0F0F0; overflow: auto; display: block; z-index: 301; padding: .5em; border: 1px solid black; margin: 0; font-weight: normal; font-style: normal; text-align: left; text-indent: 0; text-transform: none; line-height: normal; letter-spacing: normal; word-spacing: normal; word-wrap: normal; white-space: nowrap; float: none; -webkit-box-sizing: content-box; -moz-box-sizing: content-box; box-sizing: content-box; box-shadow: 5px 5px 15px #AAAAAA; -webkit-box-shadow: 5px 5px 15px #AAAAAA; -moz-box-shadow: 5px 5px 15px #AAAAAA; -khtml-box-shadow: 5px 5px 15px #AAAAAA; filter: progid:DXImageTransform.Microsoft.dropshadow(OffX=2, OffY=2, Color='gray', Positive='true')}
#MathJax_ZoomOverlay {position: absolute; left: 0; top: 0; z-index: 300; display: inline-block; width: 100%; height: 100%; border: 0; padding: 0; margin: 0; background-color: white; opacity: 0; filter: alpha(opacity=0)}
#MathJax_ZoomFrame {position: relative; display: inline-block; height: 0; width: 0}
#MathJax_ZoomEventTrap {position: absolute; left: 0; top: 0; z-index: 302; display: inline-block; border: 0; padding: 0; margin: 0; background-color: white; opacity: 0; filter: alpha(opacity=0)}
</style><style type="text/css">.MathJax_Preview {color: #888}
#MathJax_Message {position: fixed; left: 1em; bottom: 1.5em; background-color: #E6E6E6; border: 1px solid #959595; margin: 0px; padding: 2px 8px; z-index: 102; color: black; font-size: 80%; width: auto; white-space: nowrap}
#MathJax_MSIE_Frame {position: absolute; top: 0; left: 0; width: 0px; z-index: 101; border: 0px; margin: 0px; padding: 0px}
.MathJax_Error {color: #CC0000; font-style: italic}
</style><style type="text/css">.MJXp-script {font-size: .8em}
.MJXp-right {-webkit-transform-origin: right; -moz-transform-origin: right; -ms-transform-origin: right; -o-transform-origin: right; transform-origin: right}
.MJXp-bold {font-weight: bold}
.MJXp-italic {font-style: italic}
.MJXp-scr {font-family: MathJax_Script,'Times New Roman',Times,STIXGeneral,serif}
.MJXp-frak {font-family: MathJax_Fraktur,'Times New Roman',Times,STIXGeneral,serif}
.MJXp-sf {font-family: MathJax_SansSerif,'Times New Roman',Times,STIXGeneral,serif}
.MJXp-cal {font-family: MathJax_Caligraphic,'Times New Roman',Times,STIXGeneral,serif}
.MJXp-mono {font-family: MathJax_Typewriter,'Times New Roman',Times,STIXGeneral,serif}
.MJXp-largeop {font-size: 150%}
.MJXp-largeop.MJXp-int {vertical-align: -.2em}
.MJXp-math {display: inline-block; line-height: 1.2; text-indent: 0; font-family: 'Times New Roman',Times,STIXGeneral,serif; white-space: nowrap; border-collapse: collapse}
.MJXp-display {display: block; text-align: center; margin: 1em 0}
.MJXp-math span {display: inline-block}
.MJXp-box {display: block!important; text-align: center}
.MJXp-box:after {content: " "}
.MJXp-rule {display: block!important; margin-top: .1em}
.MJXp-char {display: block!important}
.MJXp-mo {margin: 0 .15em}
.MJXp-mfrac {margin: 0 .125em; vertical-align: .25em}
.MJXp-denom {display: inline-table!important; width: 100%}
.MJXp-denom > * {display: table-row!important}
.MJXp-surd {vertical-align: top}
.MJXp-surd > * {display: block!important}
.MJXp-script-box > *  {display: table!important; height: 50%}
.MJXp-script-box > * > * {display: table-cell!important; vertical-align: top}
.MJXp-script-box > *:last-child > * {vertical-align: bottom}
.MJXp-script-box > * > * > * {display: block!important}
.MJXp-mphantom {visibility: hidden}
.MJXp-munderover {display: inline-table!important}
.MJXp-over {display: inline-block!important; text-align: center}
.MJXp-over > * {display: block!important}
.MJXp-munderover > * {display: table-row!important}
.MJXp-mtable {vertical-align: .25em; margin: 0 .125em}
.MJXp-mtable > * {display: inline-table!important; vertical-align: middle}
.MJXp-mtr {display: table-row!important}
.MJXp-mtd {display: table-cell!important; text-align: center; padding: .5em 0 0 .5em}
.MJXp-mtr > .MJXp-mtd:first-child {padding-left: 0}
.MJXp-mtr:first-child > .MJXp-mtd {padding-top: 0}
.MJXp-mlabeledtr {display: table-row!important}
.MJXp-mlabeledtr > .MJXp-mtd:first-child {padding-left: 0}
.MJXp-mlabeledtr:first-child > .MJXp-mtd {padding-top: 0}
.MJXp-merror {background-color: #FFFF88; color: #CC0000; border: 1px solid #CC0000; padding: 1px 3px; font-style: normal; font-size: 90%}
.MJXp-scale0 {-webkit-transform: scaleX(.0); -moz-transform: scaleX(.0); -ms-transform: scaleX(.0); -o-transform: scaleX(.0); transform: scaleX(.0)}
.MJXp-scale1 {-webkit-transform: scaleX(.1); -moz-transform: scaleX(.1); -ms-transform: scaleX(.1); -o-transform: scaleX(.1); transform: scaleX(.1)}
.MJXp-scale2 {-webkit-transform: scaleX(.2); -moz-transform: scaleX(.2); -ms-transform: scaleX(.2); -o-transform: scaleX(.2); transform: scaleX(.2)}
.MJXp-scale3 {-webkit-transform: scaleX(.3); -moz-transform: scaleX(.3); -ms-transform: scaleX(.3); -o-transform: scaleX(.3); transform: scaleX(.3)}
.MJXp-scale4 {-webkit-transform: scaleX(.4); -moz-transform: scaleX(.4); -ms-transform: scaleX(.4); -o-transform: scaleX(.4); transform: scaleX(.4)}
.MJXp-scale5 {-webkit-transform: scaleX(.5); -moz-transform: scaleX(.5); -ms-transform: scaleX(.5); -o-transform: scaleX(.5); transform: scaleX(.5)}
.MJXp-scale6 {-webkit-transform: scaleX(.6); -moz-transform: scaleX(.6); -ms-transform: scaleX(.6); -o-transform: scaleX(.6); transform: scaleX(.6)}
.MJXp-scale7 {-webkit-transform: scaleX(.7); -moz-transform: scaleX(.7); -ms-transform: scaleX(.7); -o-transform: scaleX(.7); transform: scaleX(.7)}
.MJXp-scale8 {-webkit-transform: scaleX(.8); -moz-transform: scaleX(.8); -ms-transform: scaleX(.8); -o-transform: scaleX(.8); transform: scaleX(.8)}
.MJXp-scale9 {-webkit-transform: scaleX(.9); -moz-transform: scaleX(.9); -ms-transform: scaleX(.9); -o-transform: scaleX(.9); transform: scaleX(.9)}
.MathJax_PHTML .noError {vertical-align: ; font-size: 90%; text-align: left; color: black; padding: 1px 3px; border: 1px solid}
</style></head> <body class="vsc-initialized"><div id="MathJax_Message" style="display: none;"></div> <nav id="dl-menu" class="dl-menuwrapper" role="navigation"> <button class="dl-trigger">Open Menu</button> <ul class="dl-menu"> <li><a href="https://jwestrob.github.io/">Home</a></li> <li> <a href="https://jwestrob.github.io/Week_2_walkthrough/#">About</a> <ul class="dl-submenu"><li class="dl-back"><a href="https://jwestrob.github.io/Week_2_walkthrough/#">back</a></li> <li> <img src="./Week 2 Walkthrough- Read Quality and Trimming – Jacob West-Roberts_files/nerd.jpg" alt="Jacob West-Roberts photo" class="author-photo"> <h4>Jacob West-Roberts</h4> <p>Moon is a minimal, one column jekyll theme for your blog.</p> </li> <li><a href="https://jwestrob.github.io/about/"><span class="btn btn-inverse">Learn More</span></a></li> <li> <a href="http://twitter.com/jwestrob" target="_blank" rel="noopener noreferrer"><i class="fa fa-fw fa-twitter-square"></i> Twitter</a> </li> <li> <a href="http://github.com/jwestrob" target="_blank" rel="noopener noreferrer"><i class="fa fa-fw fa-github"></i> Github</a> </li> </ul><!-- /.dl-submenu --> </li> <li> <a href="https://jwestrob.github.io/Week_2_walkthrough/#">Posts</a> <ul class="dl-submenu"><li class="dl-back"><a href="https://jwestrob.github.io/Week_2_walkthrough/#">back</a></li> <li><a href="https://jwestrob.github.io/posts/">All Posts</a></li> <li><a href="https://jwestrob.github.io/tags/">All Tags</a></li> </ul> </li> <li><a href="https://jwestrob.github.io/ESPM_112L/">ESPM_112L</a></li> <li><a href="https://jwestrob.github.io/ESPM_112L_2021/">ESPM_112L_2021</a></li> </ul><!-- /.dl-menu --> </nav><!-- /.dl-menuwrapper --> <!-- Header --> <header class="header" role="banner"> <div class="wrapper animated fadeIn"> <div class="content"> <div class="post-title "> <h1>Week 2 Walkthrough- Read Quality and Trimming</h1> <h4>24 Jan 2023</h4> <a class="btn zoombtn" href="https://jwestrob.github.io/projects/"> <i class="fa fa-chevron-left"></i> </a> </div> <h1>Hello and welcome to week 2 of ESPM 112L-</h1> <h1>Metagenomic Data Analysis Lab!</h1> <ul id="markdown-toc"> <li><a href="https://jwestrob.github.io/Week_2_walkthrough/#tasks-for-today" id="markdown-toc-tasks-for-today">Tasks for Today</a></li> <li><a href="https://jwestrob.github.io/Week_2_walkthrough/#1-connecting-to-the-class-server" id="markdown-toc-1-connecting-to-the-class-server">1. Connecting to the class server</a> <ul> <li><a href="https://jwestrob.github.io/Week_2_walkthrough/#ssh" id="markdown-toc-ssh">SSH</a></li> <li><a href="https://jwestrob.github.io/Week_2_walkthrough/#changing-the-password" id="markdown-toc-changing-the-password">Changing the password</a></li> <li><a href="https://jwestrob.github.io/Week_2_walkthrough/#samples" id="markdown-toc-samples">Samples</a></li> </ul> </li> <li><a href="https://jwestrob.github.io/Week_2_walkthrough/#2-getting-your-reads" id="markdown-toc-2-getting-your-reads">2. Getting your reads</a></li> <li><a href="https://jwestrob.github.io/Week_2_walkthrough/#running-fastqc" id="markdown-toc-running-fastqc">Running FastQC</a></li> <li><a href="https://jwestrob.github.io/Week_2_walkthrough/#how-to-download-files-from-the-cluster-important" id="markdown-toc-how-to-download-files-from-the-cluster-important">How to Download Files From The Cluster (IMPORTANT)</a></li> <li><a href="https://jwestrob.github.io/Week_2_walkthrough/#trimming-with-sickle" id="markdown-toc-trimming-with-sickle">Trimming with Sickle</a></li> <li><a href="https://jwestrob.github.io/Week_2_walkthrough/#final-step-fastqc-on-the-trimmed-reads" id="markdown-toc-final-step-fastqc-on-the-trimmed-reads">Final step: FastQC on the trimmed reads</a></li> <li><a href="https://jwestrob.github.io/Week_2_walkthrough/#questions" id="markdown-toc-questions">Questions</a></li> </ul> <h3 id="tasks-for-today">Tasks for Today</h3> <p>Lab today will hopefully be easy on the technical front, and more about interpreting raw sequence data (what you receive back from a sequencing facility after sending them your extracted DNA).</p> <p>To get your points for the day, please provide the following in the submission assignment on bCourses:</p> <ul> <li>The HTML output of FastQC run before and after trimming</li> <li>Answers to questions (located at the end of the walkthrough)</li> <li>Names of your group members</li> </ul> <h3 id="1-connecting-to-the-class-server">1. Connecting to the class server</h3> <p>The class server is a computer we’re using to host your data and perform computational tasks this semester. Each of you will register for an account today, and learn how to access it. Physically, it exists just off campus in the UC Berkeley data center on Hearst; luckily for you, we can access it from anywhere. Let’s go over how.</p> <h4 id="ssh">SSH</h4> <p>SSH stands for ‘Secure Shell’. Remember how the terminal is also called a shell? We’re just going to be connecting to a new terminal session on the class server over the internet. This is one of two most important commands to remember for this class, so write it down somewhere and remember you can always come back here to see it again.</p> <p><code class="language-plaintext highlighter-rouge">ssh username@class.ggkbase.berkeley.edu</code></p> <p>Pretty simple! I will be giving out usernames in class, as well as telling you the default password. (Information security 101: Don’t give away passwords willy-nilly.) Once you’ve connected successfully, you’re going to need to change that default password.</p> <h4 id="changing-the-password">Changing the password</h4> <p>The command to change your password is <code class="language-plaintext highlighter-rouge">passwd</code>. Type this, and you will see a prompt asking for your current password. Enter it, then enter the new password- then you’re done!</p> <h4 id="samples">Samples</h4> <p>Each group will be assigned a sample today, and starting this week you’ll work with that sample. These samples are big, though, and so we’re going to have to take a couple precautions to avoid copying them and using up a ton of disk space!</p> <h3 id="2-getting-your-reads">2. Getting your reads</h3> <p>Make a folder in your home directory (use the <code class="language-plaintext highlighter-rouge">mkdir</code> command) called <code class="language-plaintext highlighter-rouge">raw.d</code>. This is how we designate the folders containing our raw sequencing data in the Banfield lab standard workflow. Then <code class="language-plaintext highlighter-rouge">cd</code> into that directory. Now let’s fetch your sequence files.</p> <p>The reads are located at <code class="language-plaintext highlighter-rouge">/class_data/reads/Cow_8_31/raw.d/Cow_8_31_trim_clean.PE.1.fastq.gz</code> and <code class="language-plaintext highlighter-rouge">/class_data/reads/Cow_8_31/raw.d/Cow_8_31_trim_clean.PE.2.fastq.gz</code>. These are called the “Forward” and “Reverse” reads, and should (almost always) be used together.</p> <p>Now that you’ve got the data loaded, let’s run our QC program, fastqc! I highly recommend one person per group do this, since it can take quite a while; this way you can get started on the rest of the lab while it’s running.</p> <h3 id="running-fastqc">Running FastQC</h3> <p>FastQC is a program to assess the quality of raw sequencing reads. We’re going to use this before and after trimming your reads to highlight the difference trimming makes.</p> <p>FastQC is loaded already onto the cluster, so you all have access. Try running the help menu with <code class="language-plaintext highlighter-rouge">fastqc -h</code>. The output should look like this:</p> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>FastQC - A high throughput sequence QC analysis tool

SYNOPSIS

fastqc seqfile1 seqfile2 .. seqfileN

fastqc [-o output dir] [--(no)extract] [-f fastq|bam|sam]
[-c contaminant file] seqfile1 .. seqfileN

DESCRIPTION

FastQC reads a set of sequence files and produces from each one a quality
control report consisting of a number of different modules, each one of
which will help to identify a different potential type of problem in your
data.

If no files to process are specified on the command line then the program
will start as an interactive graphical application.  If files are provided
on the command line then the program will run with no user interaction
required.  In this mode it is suitable for inclusion into a standardised
analysis pipeline.

The options for the program as as follows:

-h --help       Print this help file and exit

-v --version    Print the version of the program and exit

-o --outdir     Create all output files in the specified output directory.
        Please note that this directory must exist as the program
        will not create it.  If this option is not set then the
        output file for each sequence file is created in the same
        directory as the sequence file which was processed.
</code></pre></div></div> <p>Read the help menu carefully. FastQC will analyze both your forward and reverse reads at once, so make sure to provide them both!</p> <p>You have the option of putting the output in a directory (by providing a value for the <code class="language-plaintext highlighter-rouge">-o</code> flag), which is tidy, or just having FastQC output everything in your current directory. I recommend making a directory to put the output in, like so:</p> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>mkdir fastQC_output
fastqc -o fastQC_output [forward_reads] [reverse_reads]
</code></pre></div></div> <p>This took about 10 minutes to run, and you should expect it to take about that long. PLEASE do not use more than one thread (the <code class="language-plaintext highlighter-rouge">-t</code> option in fastqc) because we do not have the resources to support everyone doing this simultaneously.</p> <p>Once this is done, you’re going to have an output .html file (either in your current directory or in the output directory you specified with <code class="language-plaintext highlighter-rouge">-o</code>). We’re going to want to download it! This file is an .html file, so you want to open it in a web browser- Firefox, Chrome, Safari, Internet Explorer 6, Netscape, whatever you want.</p> <h3 id="how-to-download-files-from-the-cluster-important">How to Download Files From The Cluster (IMPORTANT)</h3> <p>This is a very important skill, and one that will come up many times throughout the semester. Historically, it’s been a bit tricky, so I want to devote some time to it.</p> <p>You have two options. The easiest of these by far if you’re on Mac or Windows is to use <a href="https://cyberduck.io/">Cyberduck</a>:</p> <ul> <li>Download using a GUI program like <a href="https://winscp.net/eng/download.php">WinSCP</a> for Windows, <a href="https://cyberduck.io/">Cyberduck</a> for Windows or Mac OSX, or Filezilla, I guess, if you’re using Linux. (Instructions on how to use Filezilla for this purpose are <a href="https://stackoverflow.com/questions/299412/is-there-any-winscp-equivalent-for-linux">here</a> although I really recommend you just use SCP if you’re on linux.)</li> </ul> <p>GUI-based SCP applications are pretty straightforward. If you have questions about them ask me in class or on slack.</p> <ul> <li> <p>Download with SCP. Here’s an example of how to do that (I also will give a demonstration in class):</p> </li> <li> <p>Get the full path to the file you want to download and copy it!</p> </li> </ul> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>   realpath example_fastqc_output.html
   # OUTPUT: /home/jwestrob/fastqc/example_fastqc_output.html
   # Copy that! ^
</code></pre></div></div> <ul> <li>Open a terminal on your computer (NOT the class server- you can leave that one open or close it) and navigate to the folder you want to download to.</li> </ul> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  cd ~/Downloads/
  scp jwestrob@class.ggkbase.berkeley.edu:/home/jwestrob/fastqc/example_fastqc_output.html .
  # In Bash, "." just means "here", i.e. your current location.                            
</code></pre></div></div> <h3 id="trimming-with-sickle">Trimming with Sickle</h3> <p>Next we’re going to use Sickle to trim your reads with <code class="language-plaintext highlighter-rouge">sickle</code>. Try running <code class="language-plaintext highlighter-rouge">sickle pe -h</code> for help (the <code class="language-plaintext highlighter-rouge">pe</code> stands for <code class="language-plaintext highlighter-rouge">paired-end</code>). This yields the following help information:</p> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>If you have separate files for forward and reverse reads:
Usage: sickle pe [options] -f &lt;paired-end forward fastq file&gt; -r &lt;paired-end reverse fastq file&gt; -t &lt;quality type&gt; -o &lt;trimmed PE forward file&gt; -p &lt;trimmed PE reverse file&gt; -s &lt;trimmed singles file&gt;

If you have one file with interleaved forward and reverse reads:
Usage: sickle pe [options] -c &lt;interleaved input file&gt; -t &lt;quality type&gt; -m &lt;interleaved trimmed paired-end output&gt; -s &lt;trimmed singles file&gt;

If you have one file with interleaved reads as input and you want ONLY one interleaved file as output:
Usage: sickle pe [options] -c &lt;interleaved input file&gt; -t &lt;quality type&gt; -M &lt;interleaved trimmed output&gt;

Options:
Paired-end separated reads
--------------------------
-f, --pe-file1, Input paired-end forward fastq file (Input files must have same number of records)
-r, --pe-file2, Input paired-end reverse fastq file
-o, --output-pe1, Output trimmed forward fastq file
-p, --output-pe2, Output trimmed reverse fastq file. Must use -s option.

Paired-end interleaved reads
----------------------------
-c, --pe-combo, Combined (interleaved) input paired-end fastq
-m, --output-combo, Output combined (interleaved) paired-end fastq file. Must use -s option.
-M, --output-combo-all, Output combined (interleaved) paired-end fastq file with any discarded read written to output file as a single N. Cannot be used with the -s option.
</code></pre></div></div> <p>There’s more, but I’ll let you read it on your terminal instead of putting it here.</p> <p>So remember, you have your forward and reverse reads, so you’ll be using the first option-</p> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>If you have separate files for forward and reverse reads:
Usage: sickle pe [options] -f &lt;paired-end forward fastq file&gt; -r &lt;paired-end reverse fastq file&gt; -t &lt;quality type&gt; -o &lt;trimmed PE forward file&gt; -p &lt;trimmed PE reverse file&gt; -s &lt;trimmed singles file&gt;
</code></pre></div></div> <p>Here’s an example, again using the example dataset. Remember not to copy paste this, because your sample name will be different!</p> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>sickle pe -t sanger -f [FORWARD READS] -r [REVERSE READS] -o Cow_8_31_trim_clean.trimmed.1.fastq.gz -p Cow_8_31_trim_clean.trimmed.2.fastq.gz -s Cow_8_31_trim_clean.trimmed.single.fastq.gz
</code></pre></div></div> <p>Notice the flag <code class="language-plaintext highlighter-rouge">-t sanger</code>; this indicates the way the per-base quality scores are formatted. This should take a couple more minutes. Also, make sure the output filenames (<code class="language-plaintext highlighter-rouge">-o</code>, <code class="language-plaintext highlighter-rouge">-p</code>, and <code class="language-plaintext highlighter-rouge">-s</code>) are different than the input filenames, so you don’t accidentally overwrite your files. (If you do, just let me know and I’ll get you the original data back.)</p> <p>This step should take about 5 minutes or so. Don’t worry if it goes longer. (Every sample is a different size.)</p> <h3 id="final-step-fastqc-on-the-trimmed-reads">Final step: FastQC on the trimmed reads</h3> <p>So now that you’ve got trimmed reads, we want to get FastQC results that show how much our quality scores have improved. Run FastQC again, this time using only the trimmed reads.</p> <p>I recommend making another directory for the trimmed FastQC output, like so:</p> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>mkdir fastQC_trimmed_output
fastqc -o fastQC_trimmed_output [trimmed_forward_reads] [trimmed_reverse_reads] [trimmed_singleton_reads]
</code></pre></div></div> <p>Note how I included the singleton reads here (i.e., reads without a mate pair). You don’t have to do this, but I recommend it!</p> <p>Now, as you did before, download the FastQC output and open it on your computer.</p> <h1 id="questions">Questions</h1> <ol> <li> <p>Do you see anything worrying in terms of quality scores in your untrimmed reads?</p> </li> <li> <p>What are the noticeable differences in the FastQC output between your trimmed and untrimmed reads?</p> </li> <li> <p>Briefly describe the pros and cons of short read vs. long read sequencing, and a project that would be appropriate (or inappropriate) to use them for.</p> </li> <li> <p>Based on the average read length and number of reads for one of your samples, answer the following question: What % of the community does a microbe with a 3,000,000 bp genome need to be at in order to be recovered at 10x coverage?</p> </li> </ol> <div class="entry-meta"> <br> <hr> <span class="entry-tags"><a href="https://jwestrob.github.io/tags/#Read%20Quality" title="Pages tagged Read Quality" class="tag"><span class="term">Read Quality</span></a></span> <span class="social-share"> <a href="https://www.facebook.com/sharer/sharer.php?u=/Week_2_walkthrough/" title="Share on Facebook" class="tag"> <span class="term"><i class="fa fa-facebook-square"></i> Share</span> </a> <a href="https://twitter.com/intent/tweet?text=/Week_2_walkthrough/" title="Share on Twitter" class="tag"> <span class="term"><i class="fa fa-twitter-square"></i> Tweet</span> </a> <a href="https://plus.google.com/share?url=/Week_2_walkthrough/" title="Share on Google+" class="tag"> <span class="term"><i class="fa fa-google-plus-square"></i> +1</span> </a> </span> <div style="clear:both"></div> </div> </div> </div> </header> <!-- JS --> <script src="./Week 2 Walkthrough- Read Quality and Trimming – Jacob West-Roberts_files/jquery-1.12.0.min.js"></script> <script src="./Week 2 Walkthrough- Read Quality and Trimming – Jacob West-Roberts_files/jquery.dlmenu.min.js"></script> <script src="./Week 2 Walkthrough- Read Quality and Trimming – Jacob West-Roberts_files/jquery.goup.min.js"></script> <script src="./Week 2 Walkthrough- Read Quality and Trimming – Jacob West-Roberts_files/jquery.magnific-popup.min.js"></script> <script src="./Week 2 Walkthrough- Read Quality and Trimming – Jacob West-Roberts_files/jquery.fitvid.min.js"></script> <script src="./Week 2 Walkthrough- Read Quality and Trimming – Jacob West-Roberts_files/scripts.js"></script> <!-- MathJax --> <script src="./Week 2 Walkthrough- Read Quality and Trimming – Jacob West-Roberts_files/MathJax.js" id=""></script>  
<div style="position: fixed; width: 40px; height: 40px; background: rgb(255, 255, 255); cursor: pointer; bottom: 10px; right: 20px; border-radius: 0px;" class="goup-container" title=""><div class="goup-arrow" style="width: 0px; height: 0px; margin: 0px auto; padding-top: 13px; border-style: solid; border-width: 0px 10px 10px; border-color: transparent transparent rgb(0, 0, 0);"></div></div></body></html>